src/routes/search-route.jsx

```diff
@@ -1,3 +1,413 @@
+      threshold: 0.2,
+    });
+    const searchArr = fuseMovieList.search(criteria?.titleDescription);
+    const searchedList = searchArr.map((item) => item.item);
+    return <MovieCardList movies={searchedList} />;
+  }
+
+  return <MovieCardList movies={sortedMovies} />;
+}
+
+/**
+ * handleMovieSort - takes in movies and criteria and returns sorted/filtered list
+ * @param {Movie[]} - movies - list of movies
+ * @param {Object} - criteria - sorting criteria object set in the useState hook
+ * @returns {Movie[]} - returns sorted list of movies
+ */
+function handleMovieSort(movies, criteria) {
+  const sortedMovies = sortMovies(movies, criteria);
+  const tomatoesRemoved = removeTomatolessFilms(sortedMovies, criteria);
+  const filteredByTomatoScore = filterTomatoScore(tomatoesRemoved, criteria);
+  const filteredByDecades = filterDecades(filteredByTomatoScore, criteria);
+  const filteredByKeyword = keywordSearch(filteredByDecades, criteria);
+  const sortedDescending = sortDescending(filteredByKeyword, criteria);
+
+  return sortedDescending;
+}
+/*
+    Relevant functions for sorting movies
+    */
+function sortMovies(movies, criteria) {
+  if (criteria?.sortingCriteria === "title") {
+    return movies.sort((a, b) => a.title.localeCompare(b.title));
+  } else if (criteria?.sortingCriteria === "popularity") {
+    return movies.sort((a, b) => b.popularity - a.popularity);
+  } else if (criteria?.sortingCriteria === "recentlyAdded") {
+    return movies.sort((a, b) => b.releaseDate - a.releaseDate);
+  }
+  return movies;
+}
+function keywordSearch(movies, criteria) {
+  const selectedKeywords = criteria?.keywords ?? [];
+  if (selectedKeywords.length < 1) {
+    return movies;
+  }
+  const filteredMovies = [];
+  for (let i = 0; i < movies.length; i++) {
+    for (let j = 0; j < movies[i]?.keywords.length; j++) {
+      for (let k = 0; k < selectedKeywords?.length; k++) {
+        if (movies[i].keywords[j]?.id == selectedKeywords[k]) {
+          filteredMovies.push(movies[i]);
+        }
+      }
+    }
+  }
+  return filteredMovies;
+}
+
+function removeTomatolessFilms(movies, criteria) {
+  if (criteria?.inclNoRt) {
+    return movies;
+  }
+  return movies.filter((movie) => typeof movie.tomatoScore === "number");
+}
+
+function filterDecades(movies, criteria) {
+  const selectedDecades = criteria?.decades ?? [];
+  if (selectedDecades.length < 1) {
+    return movies;
+  }
+  const filteredMovies = [];
+  for (let i = 0; i < movies.length; i++) {
+    for (let j = 0; j < selectedDecades.length; j++) {
+      if (
+        movies[i].releaseYear >= Number(selectedDecades[j]) &&
+        movies[i].releaseYear <= Number(selectedDecades[j]) + 9
+      ) {
+        filteredMovies.push(movies[i]);
+      }
+    }
+  }
+  return filteredMovies;
+}
+
+function filterTomatoScore(movies, criteria) {
+  if (criteria?.inclNoRt) {
+    return movies;
+  }
+  const aboveMin = movies.filter(
+    (movie) => movie.tomatoScore >= criteria?.rtMin
```

Comment:
We can combine both min & max in the same `.filter` to keep from looping through the whole array twice.